tests/test_dummy_animal_module.py

```diff
@@ -3,59 +3,53 @@
 This module tests the functionality of dummy_animal_module.py
 """
 
-# Problems:
-# handling string returns for details
-# handling initialization of animals with zero energy
-# test that eat can't drive energy negative
+import pytest
 
-import virtual_rainforest.animals.dummy_animal_module as am
 
+@pytest.fixture
+def plant_instance():
+    """Fixutre for a plant community used in tests."""
+    from virtual_rainforest.models.animals.dummy_animal_module import PlantCommunity
 
-class TestPlant:
+    return PlantCommunity(10.0, 1)
+
+
+class TestPlantCommunity:
     """Test Plant class."""
 
-    def test_plant_growth0(self):
-        """Testing plant_growth at 100% energy."""
-        p = am.Plant("tree", 10.0, 1)
-        print("testing plant growth : plant")
-        p.plant_growth()
-        assert p.energy == 1000
-
-    def test_plant_growth1(self):
-        """Testing plant_growth at 100% energy."""
-        p = am.Plant("tree", 10.0, 1)
-        print("testing plant growth : plant")
-        p.plant_growth()
-        assert p.energy == 1000
-
-    def test_plant_growth2(self):
-        """Testing plant_growth at 50% energy."""
-        p = am.Plant("tree", 10.0, 1)
-        print("testing plant growth : plant")
-        p.energy = 500
-        p.plant_growth()
-        assert p.energy == 750
-
-    def test_plant_growth3(self):
-        """Testing plant_growth at 0% energy."""
-        p = am.Plant("tree", 10.0, 1)
-        print("testing plant growth : plant")
-        p.energy = 0
-        p.plant_growth()
-        assert p.energy == 0
-
-    def test_plant_death(self):
-        """Testing plant_death."""
-        p = am.Plant("tree", 10.0, 1)
-        print("testing plant growth : plant")
-        p.plant_death()
-        assert p.alive == "dead"
+    @pytest.mark.parametrize(
+        "initial, final", [(10000.0, 10000.0), (500.0, 975.0), (0.0, 0.0)]
+    )
+    def test_grow(self, plant_instance, initial, final):
+        """Testing grow at 100%, 50%, and 0% maximum energy."""
+        plant_instance.energy = initial
+        plant_instance.grow()
+        assert plant_instance.energy == final
+
+    def test_die(self, plant_instance):
+        """Testing die."""
+        if plant_instance.is_alive:
```

Comment:
Not sure why you have the if statement here... `plant_instance` should be newly created in which case it will be alive. You probably want to `assert plant_instance.is_alive` to check this.

Reply:
In hindsight, it seems obvious that's what you meant!